Add `files` linting rules for track-level `config.json` (#155)
exercism/docs: building/configlet/lint.md

- The `"online_editor.indent_size"` value must be an integer >= 0 and <= 8
- The `"online_editor.highlightjs_language"` key is required
- The `"online_editor.highlightjs_language"` value must be a non-blank string¹

> - The `"files"` key is optional

Reviewer: ```suggestion
- The `"files"` key is required
```

- The `"files"` value must be an object

> - The `"files.solution`" key is optional

Reviewer: ```suggestion
- The `"files.solution`" key is required
```

> - The `"files.solution`" value must be an array

Reviewer: ```suggestion
- The `"files.solution`" value must be a non-empty array
```

- The `"files.solution`" values must be valid patterns⁴
- The `"files.solution`" values must not have duplicates

> - The `"files.test`" key is optional

Reviewer: ```suggestion
- The `"files.test`" key is required
```

> - The `"files.test`" value must be an array

Reviewer: ```suggestion
- The `"files.test`" value must be a non-empty array
```

- The `"files.test`" values must be valid patterns⁴
- The `"files.test`" values must not have duplicates

> - The `"files.example`" key is optional

Reviewer: ```suggestion
- The `"files.example`" key is required if the exercise is a practice exercise
```

> - The `"files.example`" value must be an array

Reviewer: ```suggestion
- The `"files.example`" value must be a non-empty array
```

- The `"files.example`" values must be valid patterns⁴
- The `"files.example`" values must not have duplicates

> - The `"files.exemplar`" key is optional

Reviewer: ```suggestion
- The `"files.exemplar`" key is required if the exercise is a concept exercise
```

- The `"files.exemplar`" value must be an array
- The `"files.exemplar`" values must be valid patterns⁴
- The `"files.exemplar`" values must not have duplicates
- The `"files.editor`" key is optional

> - The `"files.editor`" value must be an array

Reviewer: Would it be clear that while the key is optional, _if_ it is present, it must be a non-empty array?

```suggestion
- The `"files.editor`" value must be a non-empty array
```
